src/views/routes/form/ServiceSelector.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { FC, Ref, useState } from 'react';
+import { Control, Controller } from 'react-hook-form';
+
+import { modelToGroupVersionKind, ServiceModel } from '@kubevirt-ui/kubevirt-api/console';
+import { IoK8sApiCoreV1Service } from '@kubevirt-ui/kubevirt-api/kubernetes/models';
+import { ResourceIcon, useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+import {
+  Alert,
+  Dropdown,
+  DropdownItem,
+  DropdownList,
+  FormGroup,
+  FormHelperText,
+  HelperText,
+  HelperTextItem,
+  MenuToggle,
+  MenuToggleElement,
+} from '@patternfly/react-core';
+import Loading from '@utils/components/Loading/Loading';
+import { useNetworkingTranslation } from '@utils/hooks/useNetworkingTranslation';
+import { getName } from '@utils/resources/shared';
+import { RouteKind } from '@utils/types';
+
+type ServiceSelectorProps = {
+  control: Control<RouteKind, any>;
+  namespace: string;
+};
+
+const ServiceGroupVersionKind = modelToGroupVersionKind(ServiceModel);
+
+const ServiceSelector: FC<ServiceSelectorProps> = ({ control, namespace }) => {
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+
+  const { t } = useNetworkingTranslation();
+
+  const [services, loaded, error] = useK8sWatchResource<IoK8sApiCoreV1Service[]>({
+    groupVersionKind: ServiceGroupVersionKind,
+    isList: true,
+    namespace,
+  });
+
+  if (!loaded) return <Loading />;
+
+  if (error) return <Alert title={t('Error')}></Alert>;
+
+  return (
+    <Controller
+      control={control}
+      name="spec.to.name"
+      render={({ field: { onChange, value } }) => (
+        <FormGroup fieldId="service" isRequired label={t('Service')}>
+          <Dropdown
+            id="spec.to.name"
+            isOpen={isDropdownOpen}
+            onOpenChange={setIsDropdownOpen}
+            onSelect={() => setIsDropdownOpen(false)}
+            selected={value}
+            toggle={(toggleRef: Ref<MenuToggleElement>) => (
+              <MenuToggle
+                id="toggle-route-service"
+                isExpanded={isDropdownOpen}
+                isFullWidth
+                onClick={() => setIsDropdownOpen(!isDropdownOpen)}
+                ref={toggleRef}
+              >
+                {value ? (
+                  <>
+                    <ResourceIcon groupVersionKind={ServiceGroupVersionKind} /> {value}
+                  </>
+                ) : (
+                  t('Select a Service')
+                )}
+              </MenuToggle>
+            )}
+          >
+            <DropdownList>
+              {services.map((service) => (
```

Comment:
Should we add a conditional chaining operator here?

Reply:
okay i'll do something